base/applications/rapps/appview.cpp

```diff
@@ -1299,7 +1299,7 @@ CAppsListView::AddApplication(CAppInfo *AppInfo, BOOL InitialCheckState)
             hIcon = ExtractIconW(hInst, szIconPath.GetString(), 0);
         }
 
-        if (!hIcon)
+        if (!hIcon || (hIcon == (HICON)(INT_PTR)1))
```

Comment:
This value seems magical. Why could it return "1"? why not 2 or any other value? Does the value "1" have a special meaning? Is this behaviour actually documented in MSDN?

Reply:
`1` is an exception. This case is noted on Jira ticket and documented on
https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-extracticonw

Would switching to `ExtractIconExW()` somewhat clarify things?

Reply:
It should be fine as it currently is, if a comment is added in the code.